packages/nextjs/hooks/scaffold-eth/useTransactor.tsx

```diff
@@ -0,0 +1,108 @@
+import { TransactionRequest, TransactionResponse, TransactionReceipt } from "@ethersproject/abstract-provider";
+import { SendTransactionResult } from "@wagmi/core";
+import { BigNumber, ethers, Signer } from "ethers";
+import { Deferrable } from "ethers/lib/utils";
+import { useSigner } from "wagmi";
+import toast from "react-hot-toast";
+import { getParsedEthersError } from "~~/components/scaffold-eth/Contract/utilsContract";
+import { toast as customToast } from "~~/components/scaffold-eth/index";
+
+const DEBUG = true;
+
+type TTransactionFunc = (
+  tx: Promise<SendTransactionResult> | Deferrable<TransactionRequest> | undefined,
+  callback?: ((_param: any) => void) | undefined,
+) => Promise<Record<string, any> | undefined>;
+
+export const useTransactor = (_signer?: Signer, gasPrice?: number): TTransactionFunc | undefined => {
+  /**
+   * ### Summary 
+    if signer is provided in params use it(this means dev wants to send a raw transaction)
+    else use the signer from connect wallet (mostly its an wagmi writeTxn)
+   */
+  let signer = _signer;
+  const { data } = useSigner();
+  if (signer === undefined && data) {
+    signer = data;
+  }
+
+  const result: TTransactionFunc = async (tx, callback) => {
+    let toastId = null;
+    let transactionReceipt: TransactionReceipt | undefined;
+    let transactionResponse: SendTransactionResult | TransactionResponse | undefined;
+    if (signer) {
```

Comment:
Instead of wrapping everything inside `if (signer)`, let's do a "guard clause".

Return if !signer. I think the code is more readable that way.

Reply:
Lol, don't know why I forgot this every time :( ...Tysm 🙌